matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/MXOlmDevice.kt

```diff
@@ -429,33 +428,34 @@ internal class MXOlmDevice @Inject constructor(
     }
 
     fun storeOutboundGroupSessionForRoom(roomId: String, sessionId: String) {
-        outboundGroupSessionStore[sessionId]?.let {
+        outboundGroupSessionCache[sessionId]?.let {
             store.storeCurrentOutboundGroupSessionForRoom(roomId, it)
         }
     }
 
-    fun restoreOutboundGroupSessionForRoom(roomId: String):  MXOutboundSessionInfo? {
+    fun restoreOutboundGroupSessionForRoom(roomId: String): MXOutboundSessionInfo? {
         val restoredOutboundGroupSession = store.getCurrentOutboundGroupSessionForRoom(roomId)
         if (restoredOutboundGroupSession != null) {
             val sessionId = restoredOutboundGroupSession.outboundGroupSession.sessionIdentifier()
-            if (!outboundGroupSessionStore.containsKey(sessionId)) {
-                outboundGroupSessionStore[sessionId] = restoredOutboundGroupSession.outboundGroupSession
-                return MXOutboundSessionInfo(
-                        sessionId = sessionId,
-                        sharedWithHelper = SharedWithHelper(roomId, sessionId, store),
-                        restoredOutboundGroupSession.creationTime
-                )
-            } else {
-                restoredOutboundGroupSession.outboundGroupSession.releaseSession()
-            }
+            // cache it
+            outboundGroupSessionCache[sessionId] = restoredOutboundGroupSession.outboundGroupSession
```

Comment:
Is it OK if `outboundGroupSessionCache[sessionId]` already contains something?

Reply:
It should not happen (as only called after launch), but it won't break (but the previous session native counter part might not be released.)